core/src/main/java/com/graphhopper/util/details/AbstractPathDetailsBuilder.java

```diff
@@ -77,7 +77,7 @@ public void endInterval(int lastIndex) {
     }
 
     public Map.Entry<String, List<PathDetail>> build() {
-        return new MapEntry(getName(), pathDetails);
+        return new MapEntry<>(getName(), pathDetails);
```

Comment:
We might also switch to using `Map.entry(getName(), pathDetails)` if we don't need mutability.

Reply:
It's using a value-based class behind the scenes with all the advantages that come with it: https://www.baeldung.com/java-value-based-classes

> this is unrelated to the PR.

Yep. I have a bit of a bad habit of pointing at Java 9+ goodies when I see them.